flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvReaderFormat.java

```diff
@@ -155,16 +169,34 @@
     private static final class Reader<R, T> implements StreamFormat.Reader<T> {
         private final MappingIterator<R> iterator;
         private final Converter<R, T, Void> converter;
+        private final boolean ignoreParseErrors;
 
-        public Reader(MappingIterator<R> iterator, Converter<R, T, Void> converter) {
+        public Reader(
+                MappingIterator<R> iterator,
+                Converter<R, T, Void> converter,
+                boolean ignoreParseErrors) {
             this.iterator = checkNotNull(iterator);
             this.converter = checkNotNull(converter);
+            this.ignoreParseErrors = ignoreParseErrors;
         }
 
         @Nullable
         @Override
         public T read() throws IOException {
-            return iterator.hasNext() ? converter.convert(iterator.next(), null) : null;
+            while (true) {
+                try {
+                    if (iterator.hasNext()) {
+                        R nextElement = iterator.next();
+                        return converter.convert(nextElement, null);
+                    } else {
+                        return null;
+                    }
+                } catch (Throwable t) {
```

Comment:
Is it necessary to catch all kinds of throwables here or is `IOException` enough? It can be dangerous to catch all kinds of exceptions i.e. if you handle the `InterruptedException` you also want to set the interrupted flag of the Thread.

Reply:
I decided not to reinvent the wheel and handled it exactly as this option is currently handled in the existing implementation to avoid surprises for the users that already rely on this functionality:
https://github.com/apache/flink/blob/16452fbb57c11418796912251152947601572106/flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDataDeserializationSchema.java#L157

Reply:
Ok let's keep it.

Reply:
But good point.